rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -803,6 +807,196 @@ private void assertEventWasEmittedWithExpectedData(byte[] expectedData) {
+        void registerBtcTransaction_forSvpFundTransaction_whenSvpPeriodIsOngoing_shouldRegisterTransactionAndUpdateSvpFundTransactionValues() throws Exception {
+            // Arrange
+            BtcTransaction svpFundTransaction = createAndSaveSvpFundTransactionUnsigned();
+            signInputs(svpFundTransaction); // a transaction trying to be registered should be signed
+            setUpForTransactionRegistration(svpFundTransaction);
+
+            // Act
+            int activeFederationUtxosSizeBeforeRegisteringTx = federationSupport.getActiveFederationBtcUTXOs().size();
+            bridgeSupport.registerBtcTransaction(rskTx, svpFundTransaction.bitcoinSerialize(), btcBlockWithPmtHeight, pmtWithTransactions.bitcoinSerialize());
+            bridgeStorageProvider.save();
+
+            // Assert
+            assertNewUtxoWasSaved(activeFederationUtxosSizeBeforeRegisteringTx);
+            assertTransactionWasProcessed(svpFundTransaction.getHash());
+            assertSvpFundTransactionValuesWereUpdated();
+        }
+
+        private void assertNewUtxoWasSaved(int activeFederationUtxosSizeBeforeRegisteringTx) {
+            assertEquals(activeFederationUtxosSizeBeforeRegisteringTx + 1, federationSupport.getActiveFederationBtcUTXOs().size());
+        }
+
+        private void assertTransactionWasProcessed(Sha256Hash transactionHash) throws IOException {
+            Optional<Long> rskBlockHeightAtWhichBtcTxWasProcessed = bridgeStorageProvider.getHeightIfBtcTxhashIsAlreadyProcessed(transactionHash);
+            assertTrue(rskBlockHeightAtWhichBtcTxWasProcessed.isPresent());
+
+            long rskExecutionBlockNumber = rskExecutionBlock.getNumber();
+            assertEquals(rskExecutionBlockNumber, rskBlockHeightAtWhichBtcTxWasProcessed.get());
+        }
+
+        private void assertSvpFundTransactionValuesWereNotUpdated() {
+            assertTrue(bridgeStorageProvider.getSvpFundTxHashUnsigned().isPresent());
+            assertFalse(bridgeStorageProvider.getSvpFundTxHashSigned().isPresent());
+        }
+
+        private void assertSvpFundTransactionValuesWereUpdated() {
+            Optional<Sha256Hash> svpFundTransactionHashSignedOpt = bridgeStorageProvider.getSvpFundTxHashSigned();
+            assertTrue(svpFundTransactionHashSignedOpt.isPresent());
+
+            assertFalse(bridgeStorageProvider.getSvpFundTxHashUnsigned().isPresent());
+        }
+
+        private BtcTransaction createAndSaveSvpFundTransactionUnsigned() {
+            BtcTransaction svpFundTransaction = new BtcTransaction(btcMainnetParams);
+            Sha256Hash parentTxHash = BitcoinTestUtils.createHash(1);
+            addInput(svpFundTransaction, parentTxHash);
+            addOutputChange(svpFundTransaction);
+            savePegoutIndex(svpFundTransaction);
+            saveSvpFundTransactionHashUnsigned(svpFundTransaction.getHash());
+
+            return svpFundTransaction;
+        }
+
+        private BtcTransaction createAndSavePegout() {
```

Comment:
Out of curiosity, for these private utility methods, are we sure they dont already exist? Or something similar?

Reply:
I checked and didn't found anything!

Reply:
All transactions creation are done manually